packages/react-router/src/useMatchingRoute.ts

```diff
@@ -0,0 +1,26 @@
+import { matchRoutes } from "react-router-dom";
+import { useRoutes } from "./useRoutes.ts";
+
+export function useMatchingRoute(locationArg: Partial<Location>) {
```

Comment:
This hook doesn't exist anymore in react-router-dom but it used to be called `useRouteMatch`. Maybe we should use the same nomenclature?

Reply:
What do you mean by it doesn't exist anymore? Isn't it the [useMatches](https://reactrouter.com/en/main/hooks/use-matches) hook?

Reply:
I meant `useRouteMatch` isn't available anymore in v6, but we should use naming similar to that since it basically performs the same thing.

Reply:
Interesting, thank you, I'll give it a try.

Reply:
Done!